### ISSUE TITLE
Converting unsigned ints to floats

### ISSUE THREAD

**Issue by def-** [Araq/Nim#1348]
Doesn't work when I want to echo to converted float immediately:
```nimrod
echo float(10'u)
```
Error:
```
min.nim(1, 11) Error: internal error: getFloat
Traceback (most recent call last)
nimrod.nim(91)           nimrod
nimrod.nim(55)           handleCmdLine
main.nim(309)            mainCommand
main.nim(74)             commandCompileToC
modules.nim(194)         compileProject
modules.nim(152)         compileModule
passes.nim(193)          processModule
passes.nim(137)          processTopLevelStmt
sem.nim(413)             myProcess
sem.nim(387)             semStmtAndGenerateGenerics
semstmts.nim(1379)       semStmt
semexprs.nim(871)        semExprNoType
semexprs.nim(2034)       semExpr
semexprs.nim(1626)       semMagic
semexprs.nim(844)        semEcho
semexprs.nim(41)         semExprWithType
semexprs.nim(2034)       semExpr
semexprs.nim(1650)       semMagic
semexprs.nim(827)        semDirectOp
semexprs.nim(820)        afterCallActions
semexprs.nim(641)        evalAtCompileTime
semfold.nim(685)         getConstExpr
semfold.nim(481)         magicCall
semfold.nim(392)         evalOp
ast.nim(1480)            getFloat
msgs.nim(837)            internalError
msgs.nim(814)            liMessage
msgs.nim(730)            handleError
FAILURE
```

**Comment by skyfex** [comment]
This appears to be fixed (using version 0.10.1)

**Comment by dom96** [comment]
Works for me on ``devel``
